test/Dialect/FIRRTL/grand-central-signal-mappings.mlir

```diff
@@ -0,0 +1,54 @@
+// RUN: circt-opt --firrtl-grand-central-signal-mappings %s | FileCheck %s
+
+firrtl.circuit "SubCircuit" {
+  firrtl.extmodule @FooExtern(in %clockIn: !firrtl.clock, out %clockOut: !firrtl.clock)
+  firrtl.extmodule @BarExtern(in %someInput: !firrtl.uint<42>, out %someOutput: !firrtl.uint<42>)
+
+  // CHECK-LABEL: firrtl.module @Foo_signal_mappings
+  // CHECK-SAME:    in %clock_sink: !firrtl.clock
+  // CHECK-SAME:    out %clock_source: !firrtl.clock
+  // CHECK-SAME:  ) {
+  // CHECK:         [[T1:%.+]] = firrtl.verbatim.wire "MainA.clock"
+  // CHECK:         firrtl.force [[T1]], %clock_sink
+  // CHECK:         [[T2:%.+]] = firrtl.verbatim.wire "MainB.clock"
+  // CHECK:         firrtl.connect %clock_source, [[T2]]
+  // CHECK:       }
```

Comment:
Related to the comments above, this really could be an SV module. This is probably okay if we always run after lower types. 🤔

Reply:
True, but that would require us to carry the annotation across `LowerTypes`, which might be more difficult than lowering to IR ops and then having them run through `LowerTypes`.

Reply:
At the end of the day it wasn't that difficult to get `LowerTypes` to properly carry annotations across and extend them with field information. That allows us to go either way here. We currently do run pretty late in the pipeline (after inlining and constprop), so this should be fine.